workflow/scripts/add_electricity.py

```diff
@@ -1231,16 +1084,196 @@ def attach_ads_renewables(n, plants_df, renewable_carriers, extendable_carriers,
+
+    intersection = set(demand.columns).intersection(n.buses.load_dissag.unique())
+    demand = demand[list(intersection)]
+    return disaggregate_demand_to_buses(n, demand)
+
+
+
+def disaggregate_demand_to_buses(n: pypsa.Network, 
+                                 demand: pd.DataFrame) -> pd.DataFrame:
+    """
+    Zone power demand is disaggregated to buses proportional to Pd,
+    where Pd is the real power demand (MW).
+    """
+    demand_per_bus_pu = (n.buses.set_index("load_dissag").Pd / n.buses.groupby("load_dissag").sum().Pd)
+    demand_per_bus = demand_per_bus_pu.multiply(demand)
+    demand_per_bus.fillna(0, inplace=True)
+    demand_per_bus.columns = n.buses.index
+    return demand_per_bus
+
+
+def add_demand_from_file(n: pypsa.Network, 
+                         fn_demand: str, 
+                         demand_type: str):
+    """
+    Add demand to network from specified configuration setting. Returns network with demand added.
+    """
+    if demand_type == "breakthrough":
+        demand_per_bus = prepare_breakthrough_demand_data(n, fn_demand)
+    elif demand_type == "ads":
+        demand_per_bus = prepare_ads_demand(n, fn_demand)
+    elif demand_type == "pypsa-usa":
+        demand_per_bus = prepare_eia_demand(n, fn_demand)
+    else:
+        raise ValueError("Invalid demand_type. Supported values are 'breakthrough', 'ads', and 'pypsa-usa'.")
+    n.madd("Load", demand_per_bus.columns, bus=demand_per_bus.columns,
+           p_set=demand_per_bus, carrier='AC')
+
+
+def test_snapshot_year(sns_year: int, load_year: int, configuration: str):
+    if sns_year != load_year:
+        raise ValueError(f"""Snapshot start year {sns_year} does not match load year {load_year}
+                          required for {configuration} configuration.
+                          Please update the snapshot start year in the config file. \n
+                            ads requires 2032 \n
+                            breakthrough requires 2016 \n
+                            \n
+                          """)
+
+#########################################
 if __name__ == "__main__":
     if "snakemake" not in globals():
         from _helpers import mock_snakemake
         snakemake = mock_snakemake("add_electricity", interconnect='western')
     configure_logging(snakemake)
 
     params = snakemake.params
-
+    configuration = snakemake.config["network_configuration"]
     n = pypsa.Network(snakemake.input.base_network)
+    n.name = configuration
+    
+    snapshot_config = snakemake.config['snapshots']
+    sns_start = pd.to_datetime(snapshot_config['start'] + ' 07:00:00') 
+    # Shifting to 7am UTC since this is the Midnight PST time. 
+    # Work around to deal with the fact that the load data is in UTC and n.snapshots is in UTC... but annual EIA file doesnt include first 7 UTC hours of the year.... 
+    # TODO: Need to combine all EIA data into one file then filter based on n.snapshots
```

Comment:
```suggestion
    sns_start = pd.to_datetime(snapshot_config['start'] + ' 07:00:00') 
    # Shifting to 7am UTC since this is the Midnight PST time. 
    # Work around to deal with the fact that the load data is in UTC and n.snapshots is in UTC... but annual EIA file doesnt include first 7 UTC hours of the year.... 
    # TODO: Need to combine all EIA data into one file then filter based on n.snapshots
```Shifting to 7am UTC since this is the Midnight PST time. Work around to deal with the fact that the load data is in UTC and n.snapshots is in UTC... but annual EIA file doesnt include first 7 UTC hours of the year.... 
TODO: Need to combine all EIA data into one file then filter based on n.snapshots